server/src/main/java/org/elasticsearch/transport/RemoteConnectionStrategy.java

```diff
@@ -111,7 +110,7 @@ public int getNumberOfChannels() {
 
     private final int maxPendingConnectionListeners;
 
-    private static final Logger logger = LogManager.getLogger(RemoteConnectionStrategy.class);
+    private final Logger logger;
```

Comment:
Maybe if we care about the actual class in the logger here, just make this instead of manually passing the logger through the constructors?:

```java
    protected Logger logger = LogManager.getLogger(getClass());
```